src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/WPIJsonDepsPlugin.groovy

```diff
@@ -0,0 +1,339 @@
+                    def slurped = slurper.parse(it)
+                    def dep = constructJsonDependency(slurped)
+                    if (dep == null) {
+                        //TODO Display an error
+                    } else {
+                        jsonExtension.dependencies << dep
+                    }
+                }
+            }
+        }
+
+         DependencySpecExtension dse = project.extensions.getByType(DependencySpecExtension)
+
+        // Add all URLs from dependencies
+        jsonExtension.dependencies.each { JsonDependency dep ->
+            dep.mavenUrls.each { url ->
+                project.repositories.maven { MavenArtifactRepository repo ->
+                    repo.url = url
+                }
+            }
+        }
+
+        project.configurations.maybeCreate('wpiJsonJni')
+
+        project.extensions.add('javaVendorLibraries', { String... ignoreLibraries ->
+            def returns = []
+            if (jsonExtension.dependencies != null) {
+                jsonExtension.dependencies.find { (!ignoreLibraries.contains(it.name) && !ignoreLibraries.contains(it.uuid)) } each { JsonDependency dep ->
+                    dep.javaDependencies.each { JavaArtifact art ->
+                        returns << "${art.groupId}:${art.artifactId}:${art.version}"
+                    }
+                }
+            }
+            return returns
+        })
+
+        project.extensions.add('jniRoboRIOVendorLibraries', { String... ignoreLibraries ->
+            def returns = []
+            if (jsonExtension.dependencies != null) {
+                jsonExtension.dependencies.find { (!ignoreLibraries.contains(it.name) && !ignoreLibraries.contains(it.uuid)) } each { JsonDependency dep ->
+                    dep.jniDependencies.find{it.validClassifiers.contains('linuxathena')}.each { JniArtifact art ->
+                        project.dependencies.add('nativeZip', "${art.groupId}:${art.artifactId}:${art.version}:${'linuxathena'}@${art.isJar ? 'jar' : 'zip'}")
+                    }
+                }
+            }
+            return returns
+        })
+
+        project.extensions.add('jniClassifierVendorLibraries', { String classifier, String... ignoreLibraries ->
+            def returns = []
+            if (jsonExtension.dependencies != null) {
+                jsonExtension.dependencies.find { (!ignoreLibraries.contains(it.name) && !ignoreLibraries.contains(it.uuid)) } each { JsonDependency dep ->
+                    dep.jniDependencies.find{it.validClassifiers.contains(classifier)}.each { JniArtifact art ->
+                        project.dependencies.add('nativeZip', "${art.groupId}:${art.artifactId}:${art.version}:${classifier}@${art.isJar ? 'jar' : 'zip'}")
+                    }
+                }
+            }
+            return returns
+        })
+
+        project.extensions.add('jniDesktopVendorLibraries', { String... ignoreLibraries ->
+            def returns = []

```

Comment:
Unused

Reply:
What's unused? Returns? I'll fix them to match what you posted above.

Reply:
The returns are never assigned to in these last 3 cases, you're adding directly to the project dependencies and so returns is always an empty array

Reply:
Ah. All these are supposed to returns things, not directly add deps. I'll fix it.